WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.kt

```diff
@@ -0,0 +1,433 @@
+package org.wordpress.android.ui.themes
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.database.DataSetObserver
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AbsListView
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.Fragment
+import com.google.android.material.elevation.ElevationOverlayProvider
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.databinding.ThemeBrowserFragmentBinding
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.ThemeModel
+import org.wordpress.android.fluxc.store.QuickStartStore
+import org.wordpress.android.fluxc.store.ThemeStore
+import org.wordpress.android.ui.ScrollableViewInitializedListener
+import org.wordpress.android.ui.plans.PlansConstants
+import org.wordpress.android.ui.quickstart.QuickStartEvent
+import org.wordpress.android.util.NetworkUtils
+import org.wordpress.android.util.QuickStartUtilsWrapper
+import org.wordpress.android.util.StringUtils
+import org.wordpress.android.util.ToastUtils
+import org.wordpress.android.util.WPSwipeToRefreshHelper
+import org.wordpress.android.util.analytics.AnalyticsUtils
+import org.wordpress.android.util.extensions.getParcelableCompat
+import org.wordpress.android.util.extensions.getSerializableCompat
+import org.wordpress.android.util.helpers.SwipeToRefreshHelper
+import org.wordpress.android.util.image.ImageManager
+import javax.inject.Inject
+
+/**
+ * A fragment display the themes on a grid view.
+ */
+class ThemeBrowserFragment : Fragment(), AbsListView.RecyclerListener,
+    SearchView.OnQueryTextListener {
+    interface ThemeBrowserFragmentCallback {
+        fun onActivateSelected(themeId: String)
+        fun onTryAndCustomizeSelected(themeId: String?)
+        fun onViewSelected(themeId: String)
+        fun onDetailsSelected(themeId: String?)
+        fun onSupportSelected(themeId: String?)
+        fun onSwipeToRefresh()
+    }
+
+    private var _binding: ThemeBrowserFragmentBinding? = null
+    private val binding get() = _binding!!
```

Comment:
I like the decision we made. I also had concerns about using `_binding!!` Updated: c4c99213a2884013a1c7b1648f9f2d302d3d557c

Reply:
Awesome @irfano , thank you! 🙇 🚀 

Just for my FYI, you used `requireNotNull(binding)` vs `binding?` on a couple of places other than `onCreateView(...)`, why is that? 🤔

I think I understand why you did that on `onResume()`, although it could have been re-written as:

```kotlin
binding?.themeListview?.id?.let {
    (activity as? ScrollableViewInitializedListener)?.onScrollableViewInitialized(it)
}
```

However, doing that on `onActivityCreated(...)` is even easier, by re-writting it to:

```
binding?.themeListview?.adapter = adapter
```

Can you help me understand? 🙏 

PS: I am just trying to understand the diff, nothing more nothing less.

Reply:
The [binding](https://developer.android.com/topic/libraries/view-binding#fragments) is only valid between onCreateView and onDestroyView. I used `requireNotNull(binding)` everywhere because all of them are within the onCreateView and onDestroyView phases of the fragment's lifecycle, except `updateDisplay()`. Because `updateDisplay()` is triggered from `ThemeBrowserAdapter`.

Reply:
Got it, nice thanks so much for explaining this @irfano , let's merge! 🙇 ❤️ 🚀